runtime/int.c

```diff
@@ -0,0 +1,20 @@
+#include <stdio.h>
+#include "balrt.h"
+
+
+// lang.int:MIN_VALUE is the longest, which is 17 chars
+#define MAX_HEX_INT_LEN 17
+
+GC TaggedPtr _Bint__toHexString(int64_t i) {
+    int size = smallStringSize(MAX_HEX_INT_LEN);
```

Comment:
I would do this differently, so that you don't allocate unnecessary space (also allow for nul byte):

```
char buf[1 + 2*8 + 1]; // allow for sign, 2 hex digits for each byte and the nul byte
len = sprintf(buf, ...)
SmallStringPtr p = _bal_alloc(len);
memcpy(p->bytes, buf, len)
```